crates/bevy_pbr/src/deferred/pbr_deferred_types.wgsl

```diff
@@ -0,0 +1,107 @@
+#define_import_path bevy_pbr::pbr_deferred_types
+#import bevy_pbr::mesh_types MESH_FLAGS_SHADOW_RECEIVER_BIT 
+#import bevy_pbr::pbr_types STANDARD_MATERIAL_FLAGS_FOG_ENABLED_BIT, STANDARD_MATERIAL_FLAGS_UNLIT_BIT
+
+// Maximum of 8 bits available
+const DEFERRED_FLAGS_UNLIT_BIT: u32                 = 1u;
+const DEFERRED_FLAGS_FOG_ENABLED_BIT: u32           = 2u;
+const DEFERRED_MESH_FLAGS_SHADOW_RECEIVER_BIT: u32  = 4u;
+
+fn deferred_flags_from_mesh_material_flags(mesh_flags: u32, mat_flags: u32) -> u32 {
+    var flags = 0u;
+    flags |= u32((mesh_flags & MESH_FLAGS_SHADOW_RECEIVER_BIT) != 0u) * DEFERRED_MESH_FLAGS_SHADOW_RECEIVER_BIT;
+    flags |= u32((mat_flags & STANDARD_MATERIAL_FLAGS_FOG_ENABLED_BIT) != 0u) * DEFERRED_FLAGS_FOG_ENABLED_BIT;
+    flags |= u32((mat_flags & STANDARD_MATERIAL_FLAGS_UNLIT_BIT) != 0u) * DEFERRED_FLAGS_UNLIT_BIT;
+    return flags;
+}
+
+fn mesh_material_flags_from_deferred_flags(deferred_flags: u32) -> vec2<u32> {
+    var mat_flags = 0u;
+    var mesh_flags = 0u;
+    mesh_flags |= u32((deferred_flags & DEFERRED_MESH_FLAGS_SHADOW_RECEIVER_BIT) != 0u) * MESH_FLAGS_SHADOW_RECEIVER_BIT;
+    mat_flags |= u32((deferred_flags & DEFERRED_FLAGS_FOG_ENABLED_BIT) != 0u) * STANDARD_MATERIAL_FLAGS_FOG_ENABLED_BIT;
+    mat_flags |= u32((deferred_flags & DEFERRED_FLAGS_UNLIT_BIT) != 0u) * STANDARD_MATERIAL_FLAGS_UNLIT_BIT;
+    return vec2(mesh_flags, mat_flags);
+}
```

Comment:
Ok, so we have to remember to update these whenever adding any standard material or mesh flags? In which case, it would be great to add loud comments to where the flags are defined to make sure things get updated in sync.